osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -93,10 +96,12 @@ public WorkingBeatmap Beatmap
                 if (beatmap == value) return;
 
                 beatmap = value;
+                cancellationSource?.Cancel();
+                cancellationSource = new CancellationTokenSource();
 
                 beatmapDifficulty?.UnbindAll();
-                beatmapDifficulty = difficultyManager.GetBindableDifficulty(beatmap.BeatmapInfo);
-                beatmapDifficulty.BindValueChanged(_ => updateDisplay());
+                beatmapDifficulty = difficultyManager.GetBindableDifficulty(beatmap.BeatmapInfo, cancellationSource.Token);
+                beatmapDifficulty.BindValueChanged(_ => Schedule(updateDisplay));
 
                 updateDisplay();
```

Comment:
Shouldn't this one also technically be scheduled for even better safety? Technically there's nothing stopping someone from setting `Beatmap` on a disposed wedge. There's also another `updateDisplay()` call registered on `ruleset.ValueChanged()` in BDL, so maybe best to wrap the entire body of `updateDisplay()` with a schedule so that it can't be ever called without one?

Probably highly theoretical, but if we're fixing anyways...

Reply:
Yeah I did consider just wrapping the whole method (and using `Scheduler.AddOnce`). The only reason I left as-is is because it allows the `LoadComponentAsync` to potentially run earlier than otherwise (one frame?) by having it inline.